src/main/java/bookstore/controller/BookController.java

```diff
@@ -0,0 +1,78 @@
+import bookstore.dto.CreateBookRequestDto;
+import bookstore.service.BookService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Book management", description = "Endpoints for managing books")
+@RequiredArgsConstructor
+@RestController
+@Validated
+@RequestMapping("/books")
+public class BookController {
+    private final BookService bookService;
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @GetMapping
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
+    }
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @GetMapping("/{id}")
+    @Operation(summary = "Get book by id", description = "Get available book by certain id")
+    public BookDto getBookById(@PathVariable @Positive Long id) {
+        return bookService.getById(id);
+    }
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    @Operation(summary = "Create a new book", description = "Create a new book")
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
+        return bookService.save(requestDto);
+    }
+
+    @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @Operation(summary = "Update book by id", description = "Change parameters")
+    public BookDto update(@PathVariable Long id,
+                          @RequestBody @Valid CreateBookRequestDto requestDto
+    ) {
+        return bookService.update(id, requestDto);
+    }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("{id}")
+    @Operation(summary = "Delete book by id", description = "Delete book")
+    public void delete(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @GetMapping("/search")
+    @Operation(summary = "Search books by requirements",
+            description = "Search book by certain requirements")
+    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters, Pageable pageable) {
```

Comment:
should you validate search parameters?

Reply:
so should you validate `BookSearchParametersDto` here?

also you forgot to add `@PageableDefault`